File: cpp/Stack/easy/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> p;
        for(int i = 0; i < s.length(); i++)
        {
            if(s[i] == '(' || s[i] == '[' || s[i] == '{')
            {
                p.push(s[i]);
            }
            else
            {
                if(p.size() == 0)
                {
                    return false;
                }
                char cur = p.top();
                p.pop();
                if((s[i] == ')' && cur == '(') || (s[i] == '}' && cur == '{') || (s[i] == ']' && cur == '['))
                {
                    
                }
                else
                {
                    return false;
                }
            }
        }
        if(p.size() == 0)
        {
            return true;
        }
        return false;
    }
};

File: java/Stack/easy/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        char[] S=s.toCharArray();
        Stack<Character> a=new <Character>Stack();
        for(char c:S){
            
            if(c==']'){
                if(a.isEmpty()){return false;}
                else{
                    if(a.peek()=='[')
                    {
                        a.pop();
                    }
                    else{return false;}
                }
            }
            else if(c=='}'){
                if(a.isEmpty()){return false;}
                else{
                    if(a.peek()=='{')
                    {
                        a.pop();
                    }
                    else{return false;}
                }
            }
            else if(c==')'){
                if(a.isEmpty()){return false;}
                else{
                    if(a.peek()=='(')
                    {
                        a.pop();
                    }
                    else{return false;}
                }
            }
            else{a.push(c);}
        
        
        
        }
        if(a.isEmpty()){return true;}
    return false;
    }
}
